iconservice/icon_service_engine.py

```diff
@@ -506,7 +526,12 @@ def after_transaction_process(
             # Synchronize the timing between I-Score calculation and term change
             self._sync_end_block_height_of_calc_and_term(context, next_term)
 
+            context.storage.meta.put_last_term_end_block(context.meta_block_batch, next_term.start_block_height - 1)
+
         if context.revision >= REV_IISS:
+            if flag & (PrecommitFlag.GENESIS_IISS_CALC | PrecommitFlag.IISS_CALC):
+                last_calc_end_block_height: int = context.storage.iiss.get_end_block_height_of_calc(context)
```

Comment:
I think Optional[int] is proper. at the time of starting IISS_REV, None data could be return